src/tangents.jl

```diff
@@ -1334,6 +1334,61 @@ for T in [Symbol, Int, Val]
     @eval increment_field!!(::NoTangent, ::NoTangent, f::Union{$T}) = NoTangent()
 end
 
+# TODO: x instead of primal?
+# TODO: a primal shouldn't be modified in-place? e.g. a vector should not be overwritten?
```

Comment:
Agreed, it should be `translate_to_primal!!(tangent, primal)` where the `tangent` can get modified (because our assumption is that we won't need it i `Tangent` form anymore) but the `primal` cannot.